src/jdk.jpackage/share/native/common/FileUtils.cpp

```diff
@@ -54,7 +53,15 @@ bool isDirSeparator(const tstring::value_type c) {
 
 
 tstring dirname(const tstring &path) {
-    tstring::size_type pos = path.find_last_of(_T("\\/"));
+    tstring::size_type pos;
+    if (tstrings::endsWith(path, _T("/."))) {
```

Comment:
I'd add `|| tstrings::endsWith(path, _T("\\."))` to cover similar case for Windows paths